AirportBooking/Services/BookingService.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AirportBooking.Models;
+using AirportBooking.Repositories;
+using AirportBooking.UI;
+
+namespace AirportBooking.Services
+{
+    public class BookingService
+    {
+        private readonly IBookingRepository _bookingRepository;
+        private readonly FlightService _flightService;
+        private readonly PassengerService _passengerService;
+
+
+        public BookingService(
+            IBookingRepository bookingRepository,
+            FlightService flightService,
+            PassengerService passengerService)
+        {
+            _bookingRepository = bookingRepository;
+            _flightService = flightService;
+            _passengerService = passengerService;
+        }
+
+        public async Task<IEnumerable<Booking>> GetAllBookingsAsync()
+        {
+            return await _bookingRepository.GetAll();
+        }
+
+        public async Task<Booking?> GetBookingByIdAsync(string id)
+        {
+            return await _bookingRepository.GetById(id);
+        }
+
+        public async Task<IEnumerable<Booking>> GetBookingsByPassengerIdAsync(string passengerId)
+        {
+            var bookings = await _bookingRepository.GetAll();
+            return bookings.Where(b => b.PassengerId == passengerId);
+        }
+
+        public async Task<IEnumerable<Booking>> SearchBookingsAsync(
+            string? flightId = null,
+            string? passengerId = null,
+            string? departureCountry = null,
+            string? destinationCountry = null,
+            DateTime? departureDate = null,
+            string? departureAirport = null,
+            string? arrivalAirport = null,
+            FlightClass? flightClass = null,
+            decimal? maxPrice = null)
+        {
+            return await _bookingRepository.SearchBookingsAsync(
+            flightId,
+            passengerId,
+            departureCountry,
+            destinationCountry,
+            departureDate,
+            departureAirport,
+            arrivalAirport,
+            flightClass,
+            maxPrice
+        );
+        }
+
+
+        public async Task<Booking> CreateBookingAsync(string passengerEmail, string flightId, string classInputStr, int numberOfSeats)
+        {
+            var passenger = await _passengerService.GetPassengerByEmailAsync(passengerEmail);
+            if (passenger == null)
+            {
+                Console.WriteLine("Passenger not found. Please enter your details to register.");
+                var name = ConsoleHelper.GetStringInput("Name: ");
```

Comment:
ok i see why this is confusing :D we shouldn't read/print anything here, this is the UI layer (Program.cs) responsibility.

You can here just throw an exception if passenger is not found, in program.cs you could catch this exception and ask the user to create the passenger then try again to book for a flight